lib/src/repository/traits/thermostat_repository.dart

```diff
@@ -34,7 +34,13 @@ class ThermostatRepository {
             trait.state.ambientTemperature.reported?.value),
       }, {
         AvailableFanModes(availableFanMode),
-        AvailableThermostatModes(availableThermostatModes)
+        AvailableThermostatModes(availableThermostatModes),
+        HeatSetPointRange(TemperatureRange(
+            min: trait.properties.heatSetPointRange.min ?? 0.0,
```

Comment:
If we use nullable double fields then now we can use null here instead of `0.0`

Reply:
Since min is marked as required, I have added exclamation so that we throw error if these come back null and handled in standard dart way.